### ISSUE TITLE
Pull requests aren'accepted

### ISSUE THREAD

**Issue by Hermholtz** [http-party/http-server#908]
I've just received a notification that my PR #872 is already more than a year old.
Pull requests are ready-made features or bugfixes that you can easily integrate with the product.
Why aren't they reviewed, accepted, rejected or closed in any other meaningful way?

**Comment by Neboer** [comment]
This project is nearly dead. Please dont care about it anymore...

**Comment by Hermholtz** [comment]
That is very unfortunate, as it is a very useful piece of software.

**Comment by eytienne** [comment]
@Neboer I saw a PR two months ago, how do you motivate your comment? any replacement in sight?

**Comment by Neboer** [comment]
Although the project's code is updating, the last release is still at Jun 1, 2022. You download it from npm registery the same version just as it was two years ago.

**Comment by eytienne** [comment]
@Neboer right, I had to clone the repo and use `yarn link` to benefit from the recent `--base-dir` feature...

**Comment by KernelDeimos** [comment]
Hi all, I took on maintenance a few months ago to help out with this, then got very busy for a couple months unfortunately. While I was doing this, I noticed that maintenance would be a lot easier if we had more technical analysis of contributions.

**If any of you want to collaborate** - Discord, Matrix, etc; I would lean toward Discord primarily because it would attract newer developers - we could then standardize these processes and bring some life back into this repo.

Steps look like this:
1. Understand the change being made and why it's being made
2. Look at existing code and note "how would you do it?" (this ensures an opportunity to bring original ideas the contributor may not have thought of)
3. Compare the solution, make sure it makes sense, etc
4. Meticulously review code for potential security issues
5. Meticulously review code for potential regressions

**Comment by Hermholtz** [comment]
Hi, it's great that it moves. While I can't engage fully at the moment, I can offer my PR https://github.com/http-party/http-server/pull/872
which was tested, and it doesn't contain any security issue. It's ready for merging and will fix a long-standing annoying issue of server not being able to proxy WebSocket connections.
Hope this great piece of software will receive adequate attention.
All the best!

**Comment by madflow** [comment]
Is there a recommended replacement?

**Comment by Hermholtz** [comment]
[Yes](https://github.com/http-party/http-server/commits/master/).

**Comment by KernelDeimos** [comment]
When I took on maintainership several months ago I was able to merge some things and do some chores, but overall I didn't have the time keep the momentum (I reached out to the original maintainers about this a few months ago). This was also when the xz vulnerability was pretty recent so I realized I had to be extra cautious because this is a relatively old package with 1276 dependents, and that makes reviewing the PRs a lot more time consuming.

@Hermholtz I merged your PR because that change is pretty easy to understand just by looking at it. If anyone has any ideas on how to make maintenance easier in general this issue is a great place to discuss those. We could ask the original maintainers to take on more maintainers from here as well perhaps.

**Comment by Hermholtz** [comment]
Thank you for merging this PR.

I think the only way to properly maintain this project is to find co-maintainers able to spend some hours on it. I am not qualified for this, as I do entirely different things, but when tasked with some ad-hoc Node.js stuff these 2 years ago, I noticed the issue with WebSockets and was able to contribute. That's all I can do at the moment, and still consider it helpful.

Hope you'll be able to find somebody to assist, as I really consider http-server to be a great package which should thrive.

**Comment by jelveh** [comment]
Hello all,

My name is Nariman Jelveh, and along with the great @KernelDeimos, we'll be helping maintain this awesome project. I'm currently going through issues one by one trying to triage them. Please rest assured that we are going to revive this great pillar of the Node.js ecosystem and community 🫡

Stay tuned!

**Comment by jelveh** [comment]
> Hi, [@KernelDeimos](https://github.com/KernelDeimos) Is it possible you publish master branch version release to npm registry?

I'll look into this. For the time being I'm going to close this issue though as it's more suitable for a discussion rather than an issue.

Please stay tuned for updates :)
